commands/helper_completion.go

```diff
@@ -0,0 +1,350 @@
+package commands
+
+import (
+	"sort"
+	"strings"
+
+	"github.com/spf13/cobra"
+
+	"github.com/MichaelMure/git-bug/bridge"
+	"github.com/MichaelMure/git-bug/bridge/core/auth"
+	"github.com/MichaelMure/git-bug/bug"
+	"github.com/MichaelMure/git-bug/cache"
+	_select "github.com/MichaelMure/git-bug/commands/select"
+)
+
+type validArgsFunction func(cmd *cobra.Command, args []string, toComplete string) (completions []string, directives cobra.ShellCompDirective)
+
+func completionHandlerError(err error) (completions []string, directives cobra.ShellCompDirective) {
+	return nil, cobra.ShellCompDirectiveError
+}
+
+func completeBridge(env *Env) validArgsFunction {
+	return func(cmd *cobra.Command, args []string, toComplete string) (completions []string, directives cobra.ShellCompDirective) {
+		if len(args) > 0 {
+			return nil, cobra.ShellCompDirectiveNoFileComp
+		}
+		if err := loadBackend(env)(cmd, args); err != nil {
+			return completionHandlerError(err)
+		}
+		defer func() {
+			_ = env.backend.Close()
+		}()
+
+		bridges, err := bridge.ConfiguredBridges(env.backend)
+		if err != nil {
+			return completionHandlerError(err)
+		}
+
+		completions = make([]string, len(bridges))
+		for i, bridge := range bridges {
+			completions[i] = bridge + "\t" + "Bridge"
+		}
+
+		return completions, cobra.ShellCompDirectiveDefault
+	}
+}
+
+func completeBridgeAuth(env *Env) validArgsFunction {
+	return func(cmd *cobra.Command, args []string, toComplete string) (completions []string, directives cobra.ShellCompDirective) {
+		if len(args) > 0 {
+			return nil, cobra.ShellCompDirectiveNoFileComp
+		}
+		if err := loadBackend(env)(cmd, args); err != nil {
+			return completionHandlerError(err)
+		}
+		defer func() {
+			_ = env.backend.Close()
+		}()
+
+		creds, err := auth.List(env.backend)
+		if err != nil {
+			return completionHandlerError(err)
+		}
+
+		completions = make([]string, len(creds))
+		for i, cred := range creds {
+			meta := make([]string, 0, len(cred.Metadata()))
+			for k, v := range cred.Metadata() {
+				meta = append(meta, k+":"+v)
+			}
+			sort.Strings(meta)
+			metaFmt := strings.Join(meta, ",")
+
+			completions[i] = cred.ID().Human() + "\t" + cred.Target() + " " + string(cred.Kind()) + " " + metaFmt
+		}
+
+		return completions, cobra.ShellCompDirectiveDefault
+	}
+}
+
+func completeBug(env *Env) validArgsFunction {
+	return func(cmd *cobra.Command, args []string, toComplete string) (completions []string, directives cobra.ShellCompDirective) {
+		if len(args) > 0 {
```

Comment:
It'd be nice to complete even if `len(args) == 0`. That's one of the problem I had, I tried to complete `git-bug select` which didn't work as your code returns an empty array.
If you did that to avoid returning thousands of completion proposition, maybe we should have an arbitrary limit?